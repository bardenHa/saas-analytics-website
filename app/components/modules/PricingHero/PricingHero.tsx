const PricingHero = () => {
  return (
    <div className="flex flex-col items-center justify-between px-4 py-12 space-y-16 text-white md:py-16 md:px-0 ">
      <div className="max-w-xl md:text-center">
        <h2 className="text-3xl font-bold md:text-5xl">Pricing Plans</h2>
        <p className="pt-6 text-sm text-secondary-text md:text-lg md:pt-6">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Elementum
          nisi aliquet volutpat pellentesque volutpat est.
        </p>
      </div>
      <div className="flex flex-col space-y-20 lg:space-y-0 lg:space-x-20 lg:flex-row justify-evenly flex-nowrap">
        <PlanCard
          title="Starter"
          excerpt="up to 3 users"
          price={29}
          theme="dark"
        />
        <PlanCard
          title="Standard"
          excerpt="up to 20 users"
          price={99}
          theme="light"
        />
        <PlanCard
          title="Premium"
          excerpt="up to 200 users"
          price={299}
          theme="dark"
        />
      </div>
    </div>
  );
};

export default PricingHero;

const PlanCard = ({
  title,
  excerpt,
  price,
  theme,
}: {
  title: string;
  excerpt: string;
  price: number;
  theme: 'dark' | 'light';
}) => {
  if (theme === 'dark') {
    return (
      <div className="flex flex-col p-12 text-center shadow-lg w-80 bg-dark-card rounded-3xl">
        <h3 className="text-3xl font-bold">{title}</h3>
        <p className="pt-2 pb-10 text-lg text-secondary-text">{excerpt}</p>
        <hr className="border-divider-color" />
        <h4 className="pt-10 text-6xl font-bold">
          <span className="text-3xl tracking-wide">£</span>
          {price}
        </h4>
        <p className="pt-2 text-secondary-text">per month</p>
        <button className="py-3 mt-10 mb-2 transition duration-150 ease-in-out transform border border-solid rounded-full hover:scale-105 hover:border-opacity-0 active:bg-primary-hover hover:bg-primary border-divider-color">
          Order
        </button>
      </div>
    );
  }
  return (
    <div className="flex flex-col p-12 text-center bg-white shadow-lg w-80 rounded-3xl">
      <h3 className="text-3xl font-bold text-primary-text">{title}</h3>
      <p className="pt-2 pb-10 text-lg text-secondary-text">{excerpt}</p>
      <hr className="border-divider-color-light" />
      <h4 className="pt-10 text-6xl font-bold text-primary-text">
        <span className="text-3xl tracking-wide">£</span>
        {price}
      </h4>
      <p className="pt-2 text-secondary-text">per month</p>
      <button className="py-3 mt-10 mb-2 transition duration-150 ease-in-out transform border border-opacity-0 border-solid rounded-full active:bg-primary-hover bg-primary border-divider-color hover:scale-105">
        Order
      </button>
    </div>
  );
};
