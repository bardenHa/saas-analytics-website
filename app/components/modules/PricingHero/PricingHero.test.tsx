import { render } from '@testing-library/react';
import PricingHero from './PricingHero';

describe('Module PricingHero', () => {
  it('should render component', () => {
    const pricingHero = render(<PricingHero />);
    expect(pricingHero).toMatchSnapshot();
  });
});
