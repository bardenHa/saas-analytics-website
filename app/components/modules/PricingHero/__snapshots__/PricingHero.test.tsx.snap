// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module PricingHero should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col items-center justify-between px-4 py-12 space-y-16 text-white lg:pb-24 md:py-16 md:px-0 "
      >
        <div
          class="max-w-xl md:text-center"
        >
          <h2
            class="text-3xl font-bold md:text-5xl"
          >
            Pricing Plans
          </h2>
          <p
            class="pt-6 text-sm text-secondary-text md:text-lg md:pt-6"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Elementum nisi aliquet volutpat pellentesque volutpat est.
          </p>
        </div>
        <div
          class="flex flex-col space-y-20 lg:space-y-0 lg:space-x-20 lg:flex-row justify-evenly flex-nowrap"
        >
          <div
            class="flex flex-col p-12 text-center shadow-lg w-80 bg-dark-card rounded-3xl"
          >
            <h3
              class="text-3xl font-bold"
            >
              Starter
            </h3>
            <p
              class="pt-2 pb-10 text-lg text-secondary-text"
            >
              up to 3 users
            </p>
            <hr
              class="border-divider-color"
            />
            <h4
              class="pt-10 text-6xl font-bold"
            >
              <span
                class="text-3xl tracking-wide"
              >
                £
              </span>
              29
            </h4>
            <p
              class="pt-2 text-secondary-text"
            >
              per month
            </p>
            <button
              class="py-3 mt-10 mb-2 transition duration-150 ease-in-out transform border border-solid rounded-full hover:scale-105 hover:border-opacity-0 active:bg-primary-hover hover:bg-primary border-divider-color"
            >
              Order
            </button>
          </div>
          <div
            class="flex flex-col p-12 text-center bg-white shadow-lg w-80 rounded-3xl"
          >
            <h3
              class="text-3xl font-bold text-primary-text"
            >
              Standard
            </h3>
            <p
              class="pt-2 pb-10 text-lg text-secondary-text"
            >
              up to 20 users
            </p>
            <hr
              class="border-divider-color-light"
            />
            <h4
              class="pt-10 text-6xl font-bold text-primary-text"
            >
              <span
                class="text-3xl tracking-wide"
              >
                £
              </span>
              99
            </h4>
            <p
              class="pt-2 text-secondary-text"
            >
              per month
            </p>
            <button
              class="py-3 mt-10 mb-2 transition duration-150 ease-in-out transform border border-opacity-0 border-solid rounded-full active:bg-primary-hover bg-primary border-divider-color hover:scale-105"
            >
              Order
            </button>
          </div>
          <div
            class="flex flex-col p-12 text-center shadow-lg w-80 bg-dark-card rounded-3xl"
          >
            <h3
              class="text-3xl font-bold"
            >
              Premium
            </h3>
            <p
              class="pt-2 pb-10 text-lg text-secondary-text"
            >
              up to 200 users
            </p>
            <hr
              class="border-divider-color"
            />
            <h4
              class="pt-10 text-6xl font-bold"
            >
              <span
                class="text-3xl tracking-wide"
              >
                £
              </span>
              299
            </h4>
            <p
              class="pt-2 text-secondary-text"
            >
              per month
            </p>
            <button
              class="py-3 mt-10 mb-2 transition duration-150 ease-in-out transform border border-solid rounded-full hover:scale-105 hover:border-opacity-0 active:bg-primary-hover hover:bg-primary border-divider-color"
            >
              Order
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col items-center justify-between px-4 py-12 space-y-16 text-white lg:pb-24 md:py-16 md:px-0 "
    >
      <div
        class="max-w-xl md:text-center"
      >
        <h2
          class="text-3xl font-bold md:text-5xl"
        >
          Pricing Plans
        </h2>
        <p
          class="pt-6 text-sm text-secondary-text md:text-lg md:pt-6"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Elementum nisi aliquet volutpat pellentesque volutpat est.
        </p>
      </div>
      <div
        class="flex flex-col space-y-20 lg:space-y-0 lg:space-x-20 lg:flex-row justify-evenly flex-nowrap"
      >
        <div
          class="flex flex-col p-12 text-center shadow-lg w-80 bg-dark-card rounded-3xl"
        >
          <h3
            class="text-3xl font-bold"
          >
            Starter
          </h3>
          <p
            class="pt-2 pb-10 text-lg text-secondary-text"
          >
            up to 3 users
          </p>
          <hr
            class="border-divider-color"
          />
          <h4
            class="pt-10 text-6xl font-bold"
          >
            <span
              class="text-3xl tracking-wide"
            >
              £
            </span>
            29
          </h4>
          <p
            class="pt-2 text-secondary-text"
          >
            per month
          </p>
          <button
            class="py-3 mt-10 mb-2 transition duration-150 ease-in-out transform border border-solid rounded-full hover:scale-105 hover:border-opacity-0 active:bg-primary-hover hover:bg-primary border-divider-color"
          >
            Order
          </button>
        </div>
        <div
          class="flex flex-col p-12 text-center bg-white shadow-lg w-80 rounded-3xl"
        >
          <h3
            class="text-3xl font-bold text-primary-text"
          >
            Standard
          </h3>
          <p
            class="pt-2 pb-10 text-lg text-secondary-text"
          >
            up to 20 users
          </p>
          <hr
            class="border-divider-color-light"
          />
          <h4
            class="pt-10 text-6xl font-bold text-primary-text"
          >
            <span
              class="text-3xl tracking-wide"
            >
              £
            </span>
            99
          </h4>
          <p
            class="pt-2 text-secondary-text"
          >
            per month
          </p>
          <button
            class="py-3 mt-10 mb-2 transition duration-150 ease-in-out transform border border-opacity-0 border-solid rounded-full active:bg-primary-hover bg-primary border-divider-color hover:scale-105"
          >
            Order
          </button>
        </div>
        <div
          class="flex flex-col p-12 text-center shadow-lg w-80 bg-dark-card rounded-3xl"
        >
          <h3
            class="text-3xl font-bold"
          >
            Premium
          </h3>
          <p
            class="pt-2 pb-10 text-lg text-secondary-text"
          >
            up to 200 users
          </p>
          <hr
            class="border-divider-color"
          />
          <h4
            class="pt-10 text-6xl font-bold"
          >
            <span
              class="text-3xl tracking-wide"
            >
              £
            </span>
            299
          </h4>
          <p
            class="pt-2 text-secondary-text"
          >
            per month
          </p>
          <button
            class="py-3 mt-10 mb-2 transition duration-150 ease-in-out transform border border-solid rounded-full hover:scale-105 hover:border-opacity-0 active:bg-primary-hover hover:bg-primary border-divider-color"
          >
            Order
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
